### PR TITLE
record adu on publishing Blocks

### LINKED ISSUE #14884: Record and store add-on ADU at the time of its block
When we block an add-on, we need to record and store its ADU at the time of block approval (if applicable).

The ADU should also be shown on the block information admin page.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #14884

### CODE REVIEW

File: src/olympia/blocklist/models.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ class Block(ModelBase):
         max_length=255, null=False, default='', db_index=True,
         db_column='kinto_id')
     submission = models.ManyToManyField('BlocklistSubmission')
+    average_daily_users = models.IntegerField(null=True)
```
Comment (muffinresearch) [inline src/olympia/blocklist/models.py:44]: Should this reflect in its name that it's a snapshot of data at a point in time?

**Review by muffinresearch — APPROVED** [review]
r+wc

File: src/olympia/blocklist/admin.py [hunk 2]
```diff
@@ -544,10 +544,12 @@ def submission_logs(self, obj):
     def blocks(self, obj):
         # Annoyingly, we don't have the full context, but we stashed blocks
         # earlier in render_change_form().
+        complete = obj.signoff_state == BlocklistSubmission.SIGNOFF_PUBLISHED
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/admin.py:547]: All the signoff state constants start `SIGNOFF_` https://github.com/mozilla/addons-server/blob/master/src/olympia/blocklist/models.py#L202:L213 so it's more just "Published".

File: src/olympia/blocklist/models.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/models.py:44]: I played around with some alternative names but couldn't get a balance between descriptive and not overly wordy.  Suggestions welcome.

Comment (eviljeff) [reply src/olympia/blocklist/models.py:44]: @wagnerand do you care as you'll likely be running queries against it?

File: src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html [hunk 3]
```diff
@@ -8,9 +8,15 @@ <h3>{% blocktrans with count=blocks|length %}{{ count }} Add-on GUIDs{% endblock
             {% if block_obj.addon %}
               <span class="addon-name">{{ block_obj.addon.name }}</span>
             {% endif %}
-            {% if block_obj.average_daily_users is not None %}
-              ({{ block_obj.average_daily_users }} {% trans 'users' %}).
+
+            {% if submission_complete|default_if_none:False %}
+              {% if block_obj.average_daily_users is not None %}
+                {% blocktrans with users=block_obj.average_daily_users %}({{ users }} users).{% endblocktrans %}
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html:14]: Personally, I don't think it's worth it as it's an admin page.  I suppose it doesn't need translating - it's just a habit with template files.

File: src/olympia/blocklist/models.py (hunk 1 above)
Comment (wagnerand) [reply src/olympia/blocklist/models.py:44]: I don't have a very strong opinion, but maybe we could rename it to something like `blocked_average_daily_users` (or use _avg_ for _average_ to make it shorter), just to make it obvious for a non-familiar user that this is not the same as `average_daily_users` in `addons`.

Comment (muffinresearch) [reply src/olympia/blocklist/models.py:44]: Here's one suggestion: `average_daily_users_snapshot` ?

Comment (eviljeff) [reply src/olympia/blocklist/models.py]: https://travis-ci.org/github/mozilla/addons-server/builds/721408326?utm_source=github_status&utm_medium=notification

File: src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html]: https://travis-ci.org/github/mozilla/addons-server/builds/721408326?utm_source=github_status&utm_medium=notification

File: src/olympia/blocklist/models.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/models.py]: https://github.com/mozilla/addons-server/pull/15359/commits/0acea19d2b73790fa767e5db02e570e9dca27667

File: src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html]: https://github.com/mozilla/addons-server/pull/15359/commits/0acea19d2b73790fa767e5db02e570e9dca27667

**Pull request merged**